lib/spack/spack/build_environment.py

```diff
@@ -525,22 +525,29 @@ def child_fun():
     carries on.
     """
 
-    def child_execution(child_connection):
+    def child_execution(child_connection, input_stream):
         try:
             setup_package(pkg, dirty=dirty)
-            function()
+            function(input_stream)
             child_connection.send([None, None, None])
         except Exception as e:
             child_connection.send([type(e), e, None])
         finally:
             child_connection.close()
 
     parent_connection, child_connection = multiprocessing.Pipe()
-    p = multiprocessing.Process(
-        target=child_execution,
-        args=(child_connection,)
-    )
-    p.start()
+    try:
+        # Forward sys.stdin to be able to activate / deactivate
+        # verbosity pressing a key at run-time
+        input_stream = os.fdopen(os.dup(sys.stdin.fileno()))
+        p = multiprocessing.Process(
+            target=child_execution,
+            args=(child_connection, input_stream)
+        )
+        p.start()
+    finally:
+        # Close the input stream in the parent process
+        input_stream.close()
```

Comment:
According to the quote above nothing. I am closing a duplicated file descriptor in the parent: the original stays open and associated with the original file table entry. Does that make sense to you?